guava/src/com/google/common/graph/StandardMutableValueGraph.java

```diff
@@ -136,17 +137,23 @@ public boolean removeNode(N node) {
       }
     }
 
-    for (N successor : connections.successors()) {
+    // Since views are returned, we need to copy the successors that will be removed.
+    // Thus we avoid modifying the underlying view while iterating over it.
+    for (N successor : ImmutableSet.copyOf(connections.successors())) {
```

Comment:
Why `ImmutableSet` here and not `ImmutableList`? Seems unnecessary to deduplicate at this point.

Reply:
Well, `connection.successors()` returns with `Set<N>`, so I would not expect further deduplication at this point. I would be though concerned about the possible reordering what could happen through a hash-based set. (It might not cause issue during this cleanup, but it is not that trivial to prove.)

Anyway I borrowed this approach from `StandardMutableNetwork.removeNode` where elements are indeed copied to `ImmutableList`, probably I just unintentionally changed the type.

Thank you, I have fixed it.